compiler/compile/OMRCompilation.cpp

```diff
@@ -2079,6 +2081,62 @@ void OMR::Compilation::verifyCFG(TR::ResolvedMethodSymbol *methodSymbol)
       }
    }
 
+
+/*
+ * \brief
+ *    Check to make sure the rdbars are anchored and anchor a rdbar if it's found unanchored
+ *
+ * \note
+ *    Ideally all optimizations should anchor a rdbar when it's created.
+ *    This API is for fixing the unanchored rdbars in case some optimizations forget to
+ *    anchor the node by mistake. Optimizations should still try to generate correct rdbar
+ *    trees instead of relying on this API.
+ */
+void OMR::Compilation::verifyAndFixRdbarAnchors()
+   {
+   TR::NodeChecklist anchoredRdbarNodes(self());
+   TR::NodeChecklist visitedNodes(self());
+   NodeTreeTopPairList unanchoredRdbarNodes(self()->trMemory()->currentStackRegion());
+   for (TR::TreeTop *tt = self()->getStartTree(); tt; tt = tt->getNextTreeTop())
+      {
+      TR::Node *node = tt->getNode();
+      if (node->getOpCodeValue() == TR::treetop ||
+          node->getOpCode().isResolveOrNullCheck() ||
+          node->getOpCode().isAnchor())
+         {
+         if (node->getFirstChild()->getOpCode().isReadBar())
+            anchoredRdbarNodes.add(node->getFirstChild());
+         }
+      verifyAndCollectUnanchoredRdbarsInSubtree(node, tt, anchoredRdbarNodes, unanchoredRdbarNodes, visitedNodes);
+      }
+
+   for (auto it = unanchoredRdbarNodes.begin(); it != unanchoredRdbarNodes.end(); it++)
+      {
+      TR::Node *node = (*it).first;
+      TR::TreeTop *tt = (*it).second;
+      tt->insertBefore(TR::TreeTop::create(self(), TR::Node::create(TR::treetop, 1, node)));
+      traceMsg(self(), "node (n%dn) %p is an unanchored readbar, anchor it now\n", node->getGlobalIndex(), node);
```

Comment:
Might be useful to also print the id of the new treetop node.

Reply:
add printing id of the new treetop node. thanks!